internal/flink/command_shell.go

```diff
@@ -70,14 +72,15 @@ func (c *command) authenticated(authenticated func(*cobra.Command, []string) err
 }
 
 func (c *command) startFlinkSqlClient(prerunner pcmd.PreRunner, cmd *cobra.Command) error {
-	if useFakeGateway {
-		return client.StartApp(
-			mock.NewFakeFlinkGatewayClient(),
-			func() error { return nil },
-			types.ApplicationOptions{
-				Context:   c.Context,
-				UserAgent: c.Version.UserAgent,
-			}, func() {})
+	if featureflags.Manager.BoolVariation("cli.flink.internal", c.Context, config.CliLaunchDarklyClient, true, false) {
+		configKeys, err := cmd.Flags().GetStringSlice("config-key")
+		if err != nil {
+			return err
+		}
+		// if config-keys were passed, we should enter local mode
+		if len(configKeys) > 0 {
+			return c.startWithLocalMode(cmd)
+		}
```

Comment:
To make this code easier to unit-test:
1. Get both flag values
2. Enter the function if both are non-empty, and pass in the values themselves, not `cmd`

Reply:
yeah makes sense!